File: app/src/main/java/me/magnum/melonds/repositories/SettingsRepository.java
package me.magnum.melonds.repositories;

import io.reactivex.Observable;
import me.magnum.melonds.model.ControllerConfiguration;
import me.magnum.melonds.model.VideoFiltering;
import me.magnum.melonds.ui.Theme;

public interface SettingsRepository {
    Theme getTheme();

    String[] getRomSearchDirectories();

    String getBiosDirectory();
    boolean showBootScreen();

    VideoFiltering getVideoFiltering();

    boolean saveNextToRomFile();
    String getSaveFileDirectory();

    ControllerConfiguration getControllerConfiguration();
    int getSoftInputOpacity();

    Observable<Theme> observeTheme();
    Observable<String[]> observeRomSearchDirectories();

    void setControllerConfiguration(ControllerConfiguration controllerConfiguration);
}

File: app/src/main/java/me/magnum/melonds/repositories/SettingsRepository.kt
package me.magnum.melonds.repositories

import android.net.Uri
import io.reactivex.Observable
import me.magnum.melonds.model.*
import me.magnum.melonds.ui.Theme

interface SettingsRepository {
    fun getTheme(): Theme

    fun getRomSearchDirectories(): Array<Uri>

    fun getBiosDirectory(): Uri?
    fun showBootScreen(): Boolean

    fun getVideoFiltering(): VideoFiltering

    fun getRomSortingMode(): SortingMode
    fun getRomSortingOrder(): SortingOrder
    fun saveNextToRomFile(): Boolean
    fun getSaveFileDirectory(): Uri?
    fun getSaveStateDirectory(rom: Rom): String

    fun getControllerConfiguration(): ControllerConfiguration
    fun showSoftInput(): Boolean
    fun getSoftInputOpacity(): Int

    fun observeTheme(): Observable<Theme>
    fun observeRomSearchDirectories(): Observable<Array<Uri>>

    fun setBiosDirectory(directoryUri: Uri)
    fun setControllerConfiguration(controllerConfiguration: ControllerConfiguration)
    fun setRomSortingMode(sortingMode: SortingMode)
    fun setRomSortingOrder(sortingOrder: SortingOrder)
}
